### PR TITLE
[Fix] json object serialization error

### LINKED ISSUE #9827: [Bug] [DataSource]  json object serialization error
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.

### What happened

 json object serialization error

### What you expected to happen

1.  json object serialization is good

### How to reproduce

1. create datasource
2. edit datasource

### Anything else

image

### Version

3.0.0-alpha

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by SnowMoon-Dev**
## Brief change log

close [ #9827](https://github.com/apache/dolphinscheduler/issues/9827)

## Verify this pull request

Manually verified the change by testing locally.

### CODE REVIEW

File: dolphinscheduler-ui-next/src/views/datasource/list/use-detail.ts [hunk 1]
```diff
@@ -50,7 +50,7 @@ export function useDetail(getFieldsValue: Function) {
     const dataSourceRes = await queryDataSource(id)
     status.loading = false
     PREV_NAME = dataSourceRes.name
-    return dataSourceRes
+    return dataSourceRes as IDataSource
```
Comment (devosend) [inline dolphinscheduler-ui-next/src/views/datasource/list/use-detail.ts]: Why did you modify it?

File: dolphinscheduler-ui-next/src/views/datasource/list/use-form.ts [hunk 2]
```diff
@@ -138,9 +139,15 @@ export function useForm(id?: number) {
...
     state.detailForm = { ...initialValues }
   }
-  const setFieldsValue = (values: object) => {
-    state.detailForm = { ...state.detailForm, ...values }
+
+  const setFieldsValue = (values: IDataSource) => {
+    state.detailForm = {
+      ...state.detailForm,
+      ...values,
+      other: JSON.stringify(values.other)
```
Comment (devosend) [inline dolphinscheduler-ui-next/src/views/datasource/list/use-form.ts]: `values.other` may be null.  You should judge it.

File: dolphinscheduler-ui-next/src/views/datasource/list/use-form.ts [hunk 3]
```diff
@@ -138,9 +139,15 @@ export function useForm(id?: number) {
...
     state.detailForm = { ...initialValues }
   }
-  const setFieldsValue = (values: object) => {
-    state.detailForm = { ...state.detailForm, ...values }
+
+  const setFieldsValue = (values: IDataSource) => {
+    state.detailForm = {
+      ...state.detailForm,
+      ...values,
+      other: values.other ? JSON.stringify(values.other) : ''
```
Comment (devosend) [inline dolphinscheduler-ui-next/src/views/datasource/list/use-form.ts]: '' will cause the placeholder to not be displayed.

**Review by devosend — APPROVED** [review]
LGTM

File: dolphinscheduler-ui-next/src/views/datasource/list/use-form.ts (hunk 3 above)
Comment (SnowMoon-Dev) [reply dolphinscheduler-ui-next/src/views/datasource/list/use-form.ts]: Thanks for the guidance

File: dolphinscheduler-ui-next/src/views/datasource/list/use-form.ts (hunk 2 above)
Comment (SnowMoon-Dev) [reply dolphinscheduler-ui-next/src/views/datasource/list/use-form.ts]: i will fix it

**Pull request merged**